coriolis_openstack_utils/actions/flavor_actions.py

```diff
@@ -0,0 +1,169 @@
+        access_list = self._source_openstack_client.nova.flavor_access.list(
+            flavor=self.payload['src_flavor_id'])
+        src_tenant_ids = [access.tenant_id for access in access_list]
+        src_tenant_names = []
+        for tenant_id in src_tenant_ids:
+            try:
+                tenant_name = self._source_openstack_client.get_project_name(
+                    tenant_id)
+                src_tenant_names.append(tenant_name)
+            except keystoneauth1.exceptions.http.Forbidden:
+                LOG.warn("Cannot fetch source tenant \"%s\"'s name for "
+                         "destination mapping." % tenant_id)
+            except keystoneauth1.exceptions.http.NotFound:
+                LOG.warn("Cannot find source tenant \"%s\" for destination "
+                         "mapping.")
+        dest_tenant_names = [CONF.destination.new_tenant_name_format %
+                             {'original': tenant_name} for tenant_name in
+                             src_tenant_names]
+        flavor_access = self._destination_openstack_client.nova.flavor_access
+        for tenant_name in dest_tenant_names:
+            try:
+                tenant_id = self._destination_openstack_client.get_project_id(
+                        tenant_name)
+                flavor_access.add_tenant_access(dest_flavor_id, tenant_id)
+                LOG.info("Adding access for tenant \"%s\" to flavor \"%s\""
+                         % (tenant_id, dest_flavor_id))
+            except keystoneauth1.exceptions.http.Forbidden:
+                LOG.warn("Cannot fetch id of destination tenant named \"%s\" "
+                         "due to lack of priviledges of current user."
+                         "No flavor access will be added to this tenant."
+                         % tenant_name)
+            except keystoneauth1.exceptions.http.NotFound:
+                LOG.warn("Cannot find destination tenant named \"%s\"."
+                         "No flavor access will be added to this tenant")
+
+    def execute_operations(self):
+        super(FlavorCreationAction, self).print_operations()
+        dest_flavor_name = self.get_new_flavor_name()
+        done = self.check_already_done()
+        if done["done"]:
+            LOG.info(
+                "Flavor named '%s' already exists.",
+                dest_flavor_name)
+            return done["result"]
+
+        LOG.info("Creating destination flavor with name '%s'" %
+                 dest_flavor_name)
+        body = self.create_flavor_body()
+        dest_flavor = self._destination_openstack_client.nova.flavors.create(
+            **body)
+        if not dest_flavor.is_public:
+            self.add_tenant_access_to_flavor(dest_flavor)
+
+        return dest_flavor.to_dict()
+
+    def cleanup(self):
+        dest_flavor = self._destination_openstack_client.nova.flavors.find(
+            name=self.get_new_flavor_name(), is_public=None)
```

Comment:
What's the reasoning behind explicitly setting `is_public=None`? (won't it have a similar semantic to `is_public=False` considering that bool(None) == False?)
(Do please test what happens with `is_public=False` before updating the PR)

Reply:
is_public=None accounts for both is_public=False(private flavors) and is_public=True(public flavors)

Reply:
Makes sense, gg.